### DISCUSSION TITLE
USB dongle not found

### DISCUSSION THREAD

**Discussion by pmol200** [Koenkk/zigbee2mqtt#10986] [Devices]
I have the following zigbee dongle: Sonoff ZigBee 3.0 USB Dongle Plus (CC2652P)
It worked smoothly all the time, until now, I switch my Pi (4B) from Sd-card to SSD. the zigbee dongle was connected to one of the 2 USB 3.0 ports. The other one is in use by my serial cable for my energy meter.

As I want to have my SSD connected to a USB 3.0 port, I moved the zigbee dongle to an USB 2.0 port. My pi is now running smoothly from my SSD, although the zigbee dongle doesn't show up since it is connected to my USB 2.0 port.

I ran a couple of commands to see if it is located, but so far it doesn't seem so.

- lsusb
- ls -l /dev/serial/by-id
- sudo dmesg | grep tty

It was connected to ttyUSB0, but the configuration doesn't work anymore.

What can i do to find the zigbee usb dongle again, so I can adjust my configuration.yaml ?

**Comment by rdorys** [comment]
Hi, did you try to use an USB extension cable ? SSD can cause a lot of interferences on usb ports.

**Reply by pmol200** [reply]
Hi,
That's a good one. Yes I use an USB extension cable, as the zigbee dongle is quite large, I have too, otherwise I cannot use all usb ports. I did move the SSD and zigbee dongle away from eachother, but unfortunately the Pi doesn't see the zigbee dongle anymore. Very strange.
Has anyone experience similar issues?

**Comment by pmol200** [comment]
I got it working now. thanks for your help.
It was mapping to the ACM0 port, which was incorrect.

**Reply by rdorys** [reply]
Non pb!
I personnaly directly map it to serial/by-id/... in the configuration file and it works.

**Reply by pmol200** [reply]
And just when everything seemed to work perfectly, after a reboot, my zigbee device cannot be found and neither can the cable connected to the P1 port of my energy meter.

I have a question about something I don't understand. I am running docker containers, also for zigbee2mqtt. when I look on my Pi, there is a configuration.yaml file, which contains the zigbee2mqtt configuration. This file contains the path to the zigbee stick, which is: /dev/ttyUSB0.

Looking at the log file when starting the zigbee2mqtt container, is tries to access the mentioned path, but it fails.

I did manage to get access very quickly into the zigbee2mqtt container, before I was kicked out again, as it restarts al the time after trying to connect to the zigbee usb device.

in the container there is also a configuration/yaml file. this file has a different path to the usb stick, which is /dev/ttyACM0. is that how it should be ? is that an automatic translation of the configuration file on the host ?

**Reply by rdorys** [reply]
Hi,
I had the same problem.
After a reboot, the PI can remap diffently usb devices over ports (ttyUSB0, ttyACM0, ttyUSB1, etc...)
To deal with it, I did in portainer (in z2m container configuration): 

![image](https://user-images.githubusercontent.com/46494654/150849811-8c0dc519-b2d8-4022-9ba5-08a0acbc4491.png)

And add

```
serial:
  port: /dev/serial/by-id/usb-1a86_USB_Serial-if00-port0
```

in my configuration file.

Can you try it ?

**Reply by Ruben40870** [reply]
I had the same issue but all I had to do was add the device in portainer. I wasn't sure how I was going to access my configuration.yaml file but luckily I didn't need to. Thanks for the help!

**Comment by RoelG** [comment]
Hi
I'm having the same issue: trying to get the Sonoff ZigBee 3.0 USB Dongle Plus to work with my Raspberry Pi 4 (8GB) running Umbrel and an external SSD connected to one of the blue USB 3 ports. But it just doesn't show up when I do the lsusb command.
When I unplug the external hdd then it shows up.
I tried using an extension cable (1 meter) and also tried using a powered USB hub. Also tried connecting the dongle to one of the USB 2 ports with either the extension cable or usb hub, but it just doesn't show up whenever the HDD is plugged in.
Anybody any other ideas?

**Reply by RealSpyKitty** [reply]
I have the same issue but I am not using any ssd's. I am running my Rpi 4 with regular SD card.

besides that I have the same setup and issues as you and I can't find any solutions to it..
